From:	Gertjan van Wingerde <gwingerde@gmail.com>
Subject: [PATCH 2.6.33 1/8] rt2x00 : modify padding location.
Date:	Wed,  6 Jan 2010 23:52:42 +0100

From: Alban Browaeys <prahal@yahoo.com>

The padding is to be added between header and payload for the only header need
padding case.

Signed-off-by: Benoit PAPILLAULT <benoit.papillault@free.fr>
Signed-off-by: Alban Browaeys <prahal@yahoo.com>
Acked-by: Gertjan van Wingerde <gwingerde@gmail.com>
Acked-by: Ivo van Doorn <ivdoorn@gmail.com>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 drivers/net/wireless/rt2x00/rt2x00queue.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/wireless/rt2x00/rt2x00queue.c b/drivers/net/wireless/rt2x00/rt2x00queue.c
index 9915a09..37a9d20 100644
--- a/drivers/net/wireless/rt2x00/rt2x00queue.c
+++ b/drivers/net/wireless/rt2x00/rt2x00queue.c
@@ -197,7 +197,7 @@ void rt2x00queue_insert_l2pad(struct sk_buff *skb, unsigned int header_length)
 		 * the payload is already properly aligned.
 		 */
 		skb_push(skb, header_align);
-		memmove(skb->data, skb->data + header_align, frame_length);
+		memmove(skb->data, skb->data + header_align, header_length);
 		skbdesc->flags |= SKBDESC_L2_PADDED;
 	} else {
 		/*
-- 
1.6.6
